Persist data

Cona7 opened the pull request.

Cona7 committed:
save and load

Cona7 committed:
corresctions persistData

Rep2 reviewed (changes requested):
Resolve Swiftlint warnings.

Try to implement data persistence using Codable. It's a newer version of NSCoding.

https://developer.apple.com/documentation/swift/codable
https://developer.apple.com/documentation/foundation/archives_and_serialization/encoding_and_decoding_custom_types

Cona7 committed:
using Codable instead NSCoding

using Carthage and Storable framework
new class MealsArray
corrected some SwiftLint warnings
